packages/mobile/src/web3/saga.ts

```diff
@@ -83,6 +83,11 @@ export function* checkWeb3SyncProgress() {
 
       yield delay(100) // wait 100ms while web3 syncs
     } catch (error) {
+      // Check if error caused by switch to zeroSyncMode
+      const switchedToZeroSyncMode = yield select(zeroSyncSelector)
+      if (switchedToZeroSyncMode) {
+        return true
+      }
```

Comment:
If we cancelled this specific saga when zeroSync is turned on, I feel that it might be a bit more robust to unintended side effects.
What do you think?

Reply:
+1

Reply:
Agreed. Unfortunately this will take some more refactoring, since `checkWeb3SyncProgress()` is called a few different places (not through a watcher where we can easily cancel). I'll address this in a separate PR since it will involve some significant changes to the web3 sync sagas